### PR TITLE
Fix binary license check issues and update jersey to 2.34 for Pulsar SQL

### PR DESCRIPTION
**Pull request by codelipenghui**
- Update jersey version to 2.34 for the Pulsar SQL to keep consistent with the other components.
- Using a consistent version(1.22) for `jakarta.activation-api`

### CODE REVIEW

**Review by jiazhai — APPROVED** [review]

File: distribution/server/src/assemble/LICENSE.bin.txt [hunk 1]
```diff
@@ -559,22 +559,20 @@ CDDL-1.1 -- licenses/LICENSE-CDDL-1.1.txt
...
-    - org.glassfish.jersey.containers-jersey-container-servlet-core-2.31.jar
-    - org.glassfish.jersey.core-jersey-client-2.31.jar
-    - org.glassfish.jersey.core-jersey-common-2.31.jar
-    - org.glassfish.jersey.core-jersey-server-2.31.jar
-    - org.glassfish.jersey.ext-jersey-entity-filtering-2.31.jar
-    - org.glassfish.jersey.media-jersey-media-jaxb-2.31.jar
-    - org.glassfish.jersey.media-jersey-media-json-jackson-2.31.jar
-    - org.glassfish.jersey.media-jersey-media-multipart-2.31.jar
-    - org.glassfish.jersey.inject-jersey-hk2-2.31.jar
+    - org.glassfish.jersey.containers-jersey-container-servlet-2.34.jar
```
Comment (merlimat) [inline distribution/server/src/assemble/LICENSE.bin.txt:562]: The log title should probably be changed into "Updating to Jersey 2.34"

File: pulsar-broker/pom.xml [hunk 2]
```diff
@@ -230,6 +230,11 @@
       <artifactId>jersey-media-json-jackson</artifactId>
     </dependency>
 
+    <dependency>
```
Comment (merlimat) [inline pulsar-broker/pom.xml:233]: Why do we need to specify the dependency here? Pinning the version in the `dependencyManagement` section in top level pom should be enough, no?

File: pulsar-client-admin/pom.xml [hunk 3]
```diff
@@ -56,6 +56,11 @@
       <artifactId>jersey-media-json-jackson</artifactId>
     </dependency>
 
+    <dependency>
+      <groupId>jakarta.activation</groupId>
+      <artifactId>jakarta.activation-api</artifactId>
+    </dependency>
```
Comment (merlimat) [inline pulsar-client-admin/pom.xml:62]: Same as above, not sure why we need to add it here

File: pulsar-broker/pom.xml (hunk 2 above)
Comment (eolivelli) [reply pulsar-broker/pom.xml:233]: O it isn't.
The dependencyManagement section is like a default configuration but it deos not mean that dependencies declared up there take precedence

**Review by eolivelli — APPROVED** [review]

File: distribution/server/src/assemble/LICENSE.bin.txt (hunk 1 above)
Comment (codelipenghui) [reply distribution/server/src/assemble/LICENSE.bin.txt:562]: Updated the PR title

File: pulsar-client-admin/pom.xml (hunk 3 above)
Comment (codelipenghui) [reply pulsar-client-admin/pom.xml:62]: I have excluded `jakarta.activation-api` in the root pom.xml of the `jersey-media-json-jackson` dependency, some components depends on `jersey-media-json-jackson` will also exclude the `jakarta.activation-api`, In this PR we just make sure Pulsar using the consistent version of `jakarta.activation-api`, not remove the dependency. So add `jakarta.activation-api` dependency for the components which depends on `jersey-media-json-jackson` before. I'm not sure if we can remove it directly.

**Review by merlimat — APPROVED** [review]

**Pull request merged**